From 887c46f2a53d054e3b7022003e7c0678652066fc Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Rapha=C3=ABl=20Hertzog?= <hertzog@debian.org>
Date: Sat, 23 Jan 2010 22:22:39 +0100
Subject: [PATCH] Skip non-files when selecting the SERIES file

When selecting the location of the SERIES file, ensure that we
don't accidentally select a non-file (for example a directory)
when there still are other locations to try.

Bug-Debian: http://bugs.debian.org/546393
Bug-Ubuntu: https://bugs.launchpad.net/ubuntu/+bug/428657
---
 quilt/scripts/patchfns.in |    6 +++---
 1 files changed, 3 insertions(+), 3 deletions(-)

diff --git a/quilt/scripts/patchfns.in b/quilt/scripts/patchfns.in
index 9528d2c..05b7c63 100644
--- a/quilt/scripts/patchfns.in
+++ b/quilt/scripts/patchfns.in
@@ -1094,13 +1094,13 @@ else
 	: ${QUILT_SERIES:=series}
 fi
 
-if [ "${QUILT_SERIES:0:1}" = / ]
+if [ "${QUILT_SERIES:0:1}" = / -a \( -f $QUILT_SERIES -o ! -e $QUILT_SERIES \) ]
 then
 	SERIES=$QUILT_SERIES
-elif [ -e $QUILT_PC/$QUILT_SERIES ]
+elif [ -f $QUILT_PC/$QUILT_SERIES ]
 then
 	SERIES=$QUILT_PC/$QUILT_SERIES
-elif [ -e $QUILT_SERIES ]
+elif [ -f $QUILT_SERIES ]
 then
 	SERIES=$QUILT_SERIES
 else
-- 
1.6.6

